lib/util/LocationStatusManager.js

```
@@ -0,0 +1,552 @@
+                    });
+                    // if an error occurs, need to retry
+                    // for now, schedule minute from now
+                    const date = new Date();
+                    date.setMinutes(date.getMinutes() + 1);
+                    this._scheduledResumeJobs[service][location] = schedule.scheduleJob(date,
+                        triggerResume.bind(this));
+                }
+                this._logger.info('location resumed', {
+                    method: 'LocationStatusManager._scheduleResumeJob',
+                    location,
+                    service,
+                });
+            });
+        }
+        this._deleteResumeJob(location, service);
+        if (new Date() > date) {
+            triggerResume.bind(this)();
+        } else {
+            this._scheduledResumeJobs[service][location] = schedule.scheduleJob(date,
+                triggerResume.bind(this));
+            }
+    }
+
+    /**
+     * Deletes the scheduled resume of service for one or more locations
+     * @param {string} service service to pause for locations
+     * @param {string[]} locations location names
+     * @param {Function} cb callback
+     * @returns {undefined}
+     */
+    deleteScheduledResumeService(service, locations, cb) {
+        const isMongo = this._mongoServices.includes(service);
+        return async.eachLimit(locations, 10, (location, next) => {
+            if (!isMongo) {
+                this._pushActionToRedis(service, location, actions.deleteSchedule);
+                return next();
+            }
+            const paused = this._locationStatusStore[location].getServicePauseStatus(service);
+            if (paused) {
+                this._deleteResumeJob(location, service);
+                this._locationStatusStore[location].setServiceResumeSchedule(service, null);
+                return this._updateServiceStatusForLocation(location, next);
+            }
+            return next();
+        }, () => {
```

What if `_updateServiceStatusForLocation` fails, should we return an error to our user?
Also, might want to make sure it returns an "arsenal type of error".